### PR TITLE
test(callbacks): checking that expected events are emitted

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

This PR simplifies to `events_test.go` unit tests and also introduces event emission checks for unit/integration tests in `ibc_middleware_test.go`

closes: #4382 

### Commit Message / Changelog Entry

```bash
test(callbacks): checking that expected events are emitted
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [x] ~Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).~
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]
LGTM. Would like to see @chatton's review

File: modules/apps/callbacks/callbacks_test.go [hunk 1]
```diff
@@ -289,3 +291,31 @@ func (s *CallbacksTestSuite) AssertHasExecutedExpectedCallbackWithFee(
...
+func GetExpectedEvent(
+	packetDataUnmarshaler porttypes.PacketDataUnmarshaler, remainingGas uint64, data []byte, srcPortID,
+	eventPortID, eventChannelID string, seq uint64, callbackType types.CallbackType, expError error,
+) (abci.Event, bool) {
+	var (
+		callbackData types.CallbackData
+		err          error
+	)
+	switch callbackType {
+	case types.CallbackTypeReceivePacket:
```
Comment (chatton) [inline modules/apps/callbacks/callbacks_test.go:305]: if the switch has a single case and a default should we just use an `if else`?

**Review by chatton — APPROVED** [review]
LGTM nice clean up thanks @srdtrk

File: modules/apps/callbacks/callbacks_test.go (hunk 1 above)
Comment (srdtrk) [reply modules/apps/callbacks/callbacks_test.go]: done

**Pull request merged**